minifi/minifi-bootstrap/src/test/java/org/apache/nifi/minifi/bootstrap/status/reporters/StatusLoggerTest.java

```diff
@@ -194,16 +209,81 @@ public void TestErrorException() throws IOException {
         Mockito.when(queryableStatusAggregator.statusReport(MOCK_QUERY)).thenThrow(ioException);
 
         statusLogger.initialize(getProperties(LogLevel.ERROR), queryableStatusAggregator);
+        statusLogger.setScheduledExecutorService(new RunOnceScheduledExecutorService());
         statusLogger.start();
 
         verify(logger, Mockito.timeout(300).atLeastOnce()).error(ENCOUNTERED_IO_EXCEPTION, ioException);
     }
 
-    private static Properties getProperties(LogLevel logLevel){
+    private static Properties getProperties(LogLevel logLevel) {
         Properties properties = new Properties();
-        properties.setProperty(REPORT_PERIOD_KEY, "100");
+        properties.setProperty(REPORT_PERIOD_KEY, "1");
         properties.setProperty(LOGGING_LEVEL_KEY, logLevel.name());
         properties.setProperty(QUERY_KEY, MOCK_QUERY);
         return properties;
     }
+
+    private static class RunOnceScheduledExecutorService extends AbstractExecutorService implements ScheduledExecutorService {
```

Comment:
Instead of implementing the interface, what about using `Executors.newSingleThreadScheduledExecutor(ThreadFactory)` with a custom `ThreadFactory` that uses `Thread.currentThread()`?  That would simplify the implementation.

Reply:
That sounds good, will change thanks!

Reply:
Hmm, looks like you can't use the current thread to schedule Runnables "after the fact", getting IllegalThreadStateExceptions

Reply:
That's unfortunate. In that case, what about just extending `ScheduledThreadPoolExecutor` and overriding the one `scheduleAtFixedRate` that `PeriodicStatusReporter` uses?